torch/csrc/jit/autodiff.cpp

```diff
@@ -149,6 +159,32 @@ static std::vector<Value*> gradientForNode(Node* node, ArrayRef<Value*> grad_val
           return returned_grads;
         }
       }
+      case aten::addmm: {
+        JIT_ASSERT(inputs.size() == 3);
+        JIT_ASSERT(outputs.size() == 1);
+        JIT_ASSERT(grads.size() == 1);
+
+        // First take the grads from the matmul
+        auto returned_grads = mm_grads({inputs[1], inputs[2]}, grads);
+
+        // Scale grads if applicable
+        auto alpha = at::Scalar(node->t(attr::alpha));
+        if (alpha.to<double>() != 1.0) {
+          for (size_t i=0; i < returned_grads.size(); ++i) {
+            returned_grads[i] = returned_grads[i] * alpha;
+          }
+        }
+
+        // Grad from the addition
+        auto beta = at::Scalar(node->t(attr::beta));
+        auto it = returned_grads.begin();
+        if (beta.to<double>() != 1.0)
+          returned_grads.insert(it, grads[0] * beta);
+        else
+          returned_grads.insert(it, grads[0]);
```

Comment:
Soo this is expected to work only under the assumption that the add part doesn't broadcast, right?